### PR TITLE
[CP-1392] Refactoring device manager flow

### PR DESCRIPTION
**Pull request by igorbezsmertnyi**
Jira: [CP-1392]

**Description**

<details>
<summary><b>Screenshots</b></summary>
// put images here
</details>

[CP-1392]: https://appnroll.atlassian.net/browse/CP-1392?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/app/src/__deprecated__/renderer/wrappers/root-wrapper.tsx [hunk 1]
```diff
@@ -162,22 +130,16 @@ const RootWrapper: FunctionComponent<Props> = ({
 
   const handleAppUpdateAvailableCheck = (): void => {
     if (!window.navigator.onLine) {
-      toggleAppUpdateAvailable(false)
+      toggleUpdateAvailable(false)
```
Comment (dkarski) [inline packages/app/src/__deprecated__/renderer/wrappers/root-wrapper.tsx:133]: which update? OS or Center? I think that would be nice to keep App as a suffix.

File: packages/app/src/crash-dump/observers/crash-dump.observer.test.ts [hunk 2]
```diff
@@ -31,11 +29,11 @@ describe("Crash Dump Observer: observe", () => {
...
-        const deviceService = {
-          on: (eventName: DeviceServiceEventName, listener: () => void) => {
-            eventEmitterMock.on(eventName, listener)
-          },
-        } as unknown as DeviceService
+        // const deviceService = {
+        //   on: (eventName: DeviceServiceEventName, listener: () => void) => {
+        //     eventEmitterMock.on(eventName, listener)
+        //   },
+        // } as unknown as DeviceService
```
Comment (dkarski) [inline packages/app/src/crash-dump/observers/crash-dump.observer.test.ts:36]: to remove?

**Review by dkarski — CHANGES_REQUESTED** [review]
Please handle the below issues from the list 🙏

Results of the integration tests
- ✅  connect to the Harmony
- ✅  disconnect the Harmony device by unplugging the USB
- ✅  run app with connected Harmony
- ❌  connect to the Pure device by passing a passcode via the Pure device (no redirect to connecting/sync/overview).
- ✅  connect to the Pure device by passing a passcode via Mudita Center
- ✅  connect to the Pure device by passing a passcode via Mudita Center, disconnect and connect again without a passcode
- ✅/ ❌  disconnect by blocking the Pure device ( ❌ -> with a lag on the overview)
- ✅  run app with connected Pure
- ❌  disconnect the Pure device during Sync and reconnect the device throw sync fail & retry sync logic doesn't work properly
- ❌  disconnect the Pure device by unplugging the USB (no redirect to onboarding). (no deterministic, I don't find reproduce path yet)
  - additionally: attach Harmony crash UI/UX logic

File: packages/app/src/device-manager/services/device-manager.service.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+import { Device } from "App/device/modules/device"
+import { PortInfo } from "App/device-manager/types"
+import { PortInfoValidator } from "App/device-manager/validators"
+import { ListenerEvent, DeviceManagerError } from "App/device-manager/constants"
+
+export class DeviceManager {
+  public currentDevice: Device | undefined
+  // AUTO DISABLED - fix me if you like :)
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  public devicesMap: Map<string, Device> = new Map()
```
Comment (dkarski) [inline packages/app/src/device-manager/services/device-manager.service.ts:21]:
```suggestion
  public devicesMap: Map<string, Device> = new Map<string, Device> ()
```

File: packages/app/src/device-manager/services/device-manager.service.ts [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  }
+
+  get devices(): Device[] {
+    const deviceList: Device[] = []
+
+    this.devicesMap.forEach((value) => {
+      deviceList.push(value)
+    })
+
+    return deviceList
```
Comment (dkarski) [inline packages/app/src/device-manager/services/device-manager.service.ts:46]: 🟢 
```suggestion
    return Array.from(this.devicesMap.values()))
```

File: packages/app/src/device-manager/services/device-manager.service.ts [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+  }
+
+  public removeDevice(path: string): void {
+    this.devicesMap.delete(path)
+
+    if (this.currentDevice?.path === path) {
+      if (this.devicesMap.size > 0) {
+        // AUTO DISABLED - fix me if you like :)
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        this.currentDevice = this.devicesMap.values().next().value
```
Comment (dkarski) [inline packages/app/src/device-manager/services/device-manager.service.ts:79]:
```suggestion
        this.currentDevice = this.devicesMap.values().next().value as Device
```

**Review by wojcik92michal — COMMENTED** [review]
Just found two small things except the problems found by @dkarski

**Review by dkarski — APPROVED** [review]
> Results of the integration tests
> - ❌ connect to the Pure device by passing a passcode via the Pure device (no redirect to connecting/sync/overview).
> - ✅/ ❌ disconnect by blocking the Pure device ( ❌ -> with a lag on the overview)
✅ run app with connected Pure
> - ❌ disconnect the Pure device during Sync and reconnect the device throw sync fail & retry sync logic doesn't work properly
> - ❌ disconnect the Pure device by unplugging the USB (no redirect to onboarding). (no deterministic, I don't find reproduce path yet)
>   - additionally: attach Harmony crash UI/UX logic

The above issues are fixed 🤞 @wojcik92michal please double check 🙏 

Note:
 - the restore & backup flow will be overwritten in the second PR, so please skip the quality of the code 😅

**Review by dkarski — APPROVED** [review]

**Pull request merged**